lib/elasticity/strategies/alias_index.rb

```diff
@@ -66,10 +66,17 @@ def remap(index_def)
             docs = @client.mget(body: { docs: id_docs }, refresh: true)["docs"]
             break if docs.empty?
 
+            # Modify document hashes to match the mapping definition so that legacy fields aren't added
+            defined_mapping_fields = index_def[:mappings][docs.first["_type"]]["properties"].keys
```

Comment:
If the mapping it not defined then this will throw a cryptic error. Let's add a check to the beginning of this method, if the mapping is not defined we should raise an error saying remap can only happen on a document that defines a mapping.

Reply:
Actually looking at this, it will blow up and show a clear error if there is no mapping defined when it tries to create the new index on https://github.com/doximity/es-elasticity/pull/53/files#diff-61da289017228864edc1d8b8579abda8R40